File: TeeNew/Line_ColorEachLine.pas
unit Line_ColorEachLine;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, Series, TeeProcs, Chart;

type
  TLineColorEachLine = class(TBaseForm)
    CheckBox1: TCheckBox;
    Series1: TLineSeries;
    CheckBox2: TCheckBox;
    procedure FormCreate(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure CheckBox2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TLineColorEachLine.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(20);

  Chart1.View3D:=False;
  Series1.ColorEachPoint:=True;
  Series1.ColorEachLine:=False;
end;

procedure TLineColorEachLine.CheckBox1Click(Sender: TObject);
begin
  Series1.ColorEachLine:=CheckBox1.Checked;
end;

procedure TLineColorEachLine.CheckBox2Click(Sender: TObject);
begin
  CheckBox1.Enabled:=CheckBox2.Checked;
  Series1.ColorEachPoint:=CheckBox2.Checked;
end;

initialization
  RegisterClass(TLineColorEachLine);
end.

File: TeeNew/Builder2006/Line_ColorEachLine.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Line_ColorEachLine.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TLineColorEachLine *LineColorEachLine;
//---------------------------------------------------------------------------
__fastcall TLineColorEachLine::TLineColorEachLine(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TLineColorEachLine::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(20);

  Chart1->View3D = false;
  Series1->ColorEachPoint = true;
  Series1->ColorEachLine = false;
}
//---------------------------------------------------------------------------

void __fastcall TLineColorEachLine::CheckBox1Click(TObject *Sender)
{
  Series1->ColorEachLine = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TLineColorEachLine::CheckBox2Click(TObject *Sender)
{
  CheckBox1->Enabled = CheckBox2->Checked;
  Series1->ColorEachPoint = CheckBox2->Checked;
}
//---------------------------------------------------------------------------

